### ISSUE TITLE
Custom eval function documentation incorrect?

### ISSUE THREAD

**Issue by LarsDu** [Microsoft/LightGBM#2117]
## Environment info

Operating System:

CPU/GPU model:

C++/Python/R version:

LightGBM version or commit hash:

## Documentation as written on: https://lightgbm.readthedocs.io/en/latest/Python-API.html#scikit-learn-api
Custom eval function expects a callable with following signatures: func(y_true, y_pred), func(y_true, y_pred, weight) or func(y_true, y_pred, weight, group) and returns (eval_name, eval_result, is_bigger_better) or list of (eval_name, eval_result, is_bigger_better):

    y_true : array-like of shape = [n_samples]
        The target values.
    y_pred : array-like of shape = [n_samples] or shape = [n_samples * n_classes] (for multi-class task)
        The predicted values.
    weight : array-like of shape = [n_samples]
        The weight of samples.
    group : array-like
        Group/query data, used for ranking task.
    eval_name : string
        The name of evaluation.
    eval_result : float
        The eval result.
    is_bigger_better : bool
        Is eval result bigger better, e.g. AUC is bigger_better.

For multi-class task, the y_pred is group by class_id first, then group by row_id. If you want to get i-th row y_pred in j-th class, the access way is y_pred[j * num_data + i].

## Steps to reproduce

I just noticed that this custom eval function documentation lists the second argument as y_pred where y_pred is presumably the thresholded predictions of the classifier. However, I believe this should actually be *y_score* (following sklearn documentation) as the values I that get passed are clearly scores (floating point values) rather than predictions (0s and 1s)

**Comment by StrikerRUS** [comment]
@LarsDu Thanks for the propose! However, `y_pred` is the standard sklearn name for `y`s obtained from your model, no matter they are probabilities, raw values or actually class labels. They are still **predicted** values.
https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics
